server/src/models/user.model.ts

```diff
@@ -0,0 +1,24 @@
+import mongoose, { Schema, Types } from 'mongoose';
+
+interface IUserModel {
+	email: string;
+	rooms_ids: Types.ObjectId[];
+	created_at: Date;
+	updated_at: Date;
+}
+
+const userSchema = new Schema<IUserModel>(
+	{
+		email: {
+			type: String,
+			unique: true,
+			required: true,
+		},
+		rooms_ids: [Schema.Types.ObjectId],
+	},
+	{
+		timestamps: true,
+	}
+);
```

Comment:
We don't need rooms_ids.

```suggestion
const userSchema = new mongoose.Schema({
    email: {
        type: String,
        required: true,
        unique: true,    // Username for identifying the user
    },
    password: {
        type: String,
        required: true,  // Hashed password
    },
    profile_img: {
        type: String,    // URL for user profile image
    },
    status: {
        type: String,
        enum: ['online', 'offline', 'busy', 'away'], // User status for presence indication
        default: 'offline',
    },
    lastSeen: {
        type: Date,       // Timestamp of the last time the user was online
    },
    contacts: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'User',  // List of the user's contacts/friends
        },
    ],
}, { timestamps: true });

```